components/org.wso2.carbon.identity.application.authenticator.biometric/src/main/java/org/wso2/carbon/identity/application/authenticator/biometric/BiometricAuthenticator.java

```diff
@@ -0,0 +1,159 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.application.authenticator.biometric;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.AbstractApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.FederatedApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authenticator.biometric.dao.impl.BiometricDAOImpl;
+import org.wso2.carbon.identity.application.authenticator.biometric.notification.handler.impl.PushNotificationSenderImpl;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Biometric Authenticator class.
+ */
+public class BiometricAuthenticator extends AbstractApplicationAuthenticator
+        implements FederatedApplicationAuthenticator {
+
+    private static final Log log = LogFactory.getLog(BiometricAuthenticator.class);
+    private static String sessionDataKey;
+    private static String randomChallenge;
+    private static String signedChallenge;
+
+    /**
+     * Get the friendly name of the Authenticator.
+     */
+    @Override
+    public String getFriendlyName() {
+
+        return BiometricAuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        signedChallenge = request.getParameter(BiometricAuthenticatorConstants.SIGNED_CHALLENGE);
+        log.info("Signed challenge from the form submitted: " + signedChallenge);
+        return signedChallenge != null;
+    }
+
+    @Override
+    public String getContextIdentifier(javax.servlet.http.HttpServletRequest request) {
+
+        sessionDataKey = request.getParameter(BiometricAuthenticatorConstants.SESSION_DATA_KEY);
+        return sessionDataKey;
+    }
+
+    /**
+     * Get the name of the Authenticator.
+     */
+    @Override
+    public String getName() {
+
+        return BiometricAuthenticatorConstants.AUTHENTICATOR_NAME;
+    }
+
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) {
+
+        AuthenticatedUser user = context.getSequenceConfig().getStepMap().get(1).getAuthenticatedUser();
```

Comment:
If this authentication is used as the third factor, still it should work, right? Hardcoding `get(1)` might break that

Reply:
Yes. I will fix it.